### PR TITLE
gRPC blocking: fix calling grpc client within blocking service

### PR DESCRIPTION
**Pull request by michalszynkiewicz**

### CODE REVIEW

File: extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/IOThreadClientInterceptor.java [hunk 1]
```diff
@@ -21,21 +34,38 @@
...
+                                onMessageCompletion.get(60, TimeUnit.SECONDS);
+                            } catch (InterruptedException | ExecutionException e) {
+                                throw new RuntimeException("`onMessage` failed or interrupted", e);
+                            } catch (TimeoutException e) {
+                                throw new RuntimeException("`onMessage` did not complete in 60 seconds");
+                            }
+                            super.onClose(status, trailers);
                         } else {
-                            fun.run();
+                            super.onClose(status, trailers);
```
Comment (cescoffier) [inline extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/IOThreadClientInterceptor.java:68]: shouldn't you enqueue close to the `context` if any?

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/IOThreadClientInterceptor.java (hunk 1 above)
Comment (michalszynkiewicz) [reply extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/IOThreadClientInterceptor.java:68]: If I do that, the newly added test gets stuck. Do you know what problem was @Andro999b fixing by making the `onClose` executed on the event loop?

Comment (Andro999b) [reply extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/IOThreadClientInterceptor.java:68]: Hi @michalszynkiewicz. There was problem that onClose and onMessage was executed on different threads. Becuse of this when you use grpc streams you may loss messages becuse onClose can be called before onMessage. You may found more details in this issue https://github.com/quarkusio/quarkus/issues/15187

Comment (michalszynkiewicz) [reply extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/IOThreadClientInterceptor.java:68]: Thanks Andrii! I'll take your reproducer and see if with my changes it still works. And then bother Clement to figure out what's the proper solution :)

Comment (Andro999b) [reply extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/IOThreadClientInterceptor.java:68]: @michalszynkiewicz reproducer that i have in the issue not really work. I uploaded correct one here: 
https://drive.google.com/file/d/16xdsSuSV4V9pQAJt_HDYoppgqsAGOGoX/view?usp=sharing

Comment (michalszynkiewicz) [reply extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/IOThreadClientInterceptor.java:68]: Do I get correctly that we have an error if the test (executed multiple times) sometimes prints this?
```
java.util.NoSuchElementException
        at io.grpc.stub.ClientCalls$BlockingResponseStream.next(ClientCalls.java:662)
        at StreamClientTest$MyConsumer.lambda$consume$1(StreamClientTest.java:57)
        at io.vertx.core.impl.ContextImpl.lambda$executeBlocking$2(ContextImpl.java:313)
        at io.vertx.core.impl.TaskQueue.run(TaskQueue.java:76)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:834)
null
true

```


With my changes I don't see this error too.

Comment (Andro999b) [reply extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/supports/IOThreadClientInterceptor.java:68]: @michalszynkiewicz Yep, correct.

**Comment by cescoffier** [comment]
@michalszynkiewicz are we good on this issue?

**Review by cescoffier — APPROVED** [review]

**Comment by michalszynkiewicz** [comment]
@cescoffier I think we are good on it :)

**Pull request merged**